plugins/wacom-usb/fu-wac-module-scaler.c

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (C) 2018 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <string.h>
+
+#include "fu-wac-common.h"
+#include "fu-wac-device.h"
+#include "fu-wac-module-scaler.h"
+
+struct _FuWacModuleScaler {
+	FuWacModule parent_instance;
+};
+
+G_DEFINE_TYPE(FuWacModuleScaler, fu_wac_module_scaler, FU_TYPE_WAC_MODULE)
+
+#define FU_WAC_MODULE_SCALER_CRC8_POLYNOMIAL 0x07
+#define FU_WAC_MODULE_SCALER_PAYLOAD_SZ	     256
+#define FU_WAC_MODULE_SCALER_START_NORMAL    0x00
+#define FU_WAC_MODULE_SCALER_START_FULLERASE 0xFE
+
+typedef struct {
+	guint8 addr[3];
+	guint8 crc;
+	guint8 cdata[FU_WAC_MODULE_SCALER_PAYLOAD_SZ];
+} FuWacModuleScalerBlockData;
+
+static guint8
+fu_wac_module_scaler_calculate_crc(const guint8 *data, gsize sz)
+{
+	guint8 crc = ~fu_crc8(data, sz);
+	return crc;
+}
```

Comment:
we only use this once, maybe just inline it into the code as `~fu_crc8(bd->cdata, sizeof(bd->cdata))`